core/Enclave/enclave_hsm.cpp

```diff
@@ -202,6 +212,207 @@ sgx_status_t enclave_create_key(ehsm_keyblob_t *cmk, size_t cmk_size)
     return ret;
 }
 
+sgx_status_t enclave_get_parameters_for_import(ehsm_keyblob_t *cmk, size_t cmk_size,
+                                               ehsm_keyspec_t keyspec,
+                                               ehsm_data_t *pubkey, size_t pubkey_size)
+{
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+
+    if (cmk->keybloblen == 0)
+    {
+        ehsm_calc_keyblob_size(keyspec, cmk->keybloblen);
+    }
+
+    RSA *rsa_keypair = NULL;
+    BIO *bio = NULL;
+    uint8_t *pem_keypair = NULL;
+    uint32_t key_size = 0;
+    BIGNUM *e = NULL;
+
+    e = BN_new();
+    if (!e)
+        goto out;
+
+    if (!BN_set_word(e, (BN_ULONG)RSA_F4))
+        goto out;
+
+    rsa_keypair = RSA_new();
+    if (rsa_keypair == NULL)
+        goto out;
+
+    switch (keyspec)
+    {
+    case EH_RSA_2048:
+        RSA_generate_key_ex(rsa_keypair, RSA_2048_KEY_BITS, e, NULL);
+        break;
+    case EH_RSA_3072:
+        RSA_generate_key_ex(rsa_keypair, RSA_3072_KEY_BITS, e, NULL);
+        break;
+    case EH_RSA_4096:
+        RSA_generate_key_ex(rsa_keypair, RSA_4096_KEY_BITS, e, NULL);
+        break;
+    default:
+        goto out;
```

Comment:
Can user get the result that keyspec is not supported?

Reply:
fix that.if keyspec is not supported, user will get  SGX_ERROR_INVALID_PARAMETER .